src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFeatures.java

```diff
@@ -542,33 +544,29 @@ public void onPlayerDeath(PlayerDeathEvent event) {
 		dead.sendMessage(ChatColor.RED + String.format("You were slain by %s at [%s %d, %d, %d]",
 				killer, loc.getWorld().getName(), loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()));
 	}
-	
+
 	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
-	public void hitGoldBlock(PlayerInteractEvent event) {
-		if (event.getAction() != Action.LEFT_CLICK_BLOCK) {
-			return;
-		}
-		Block block = event.getClickedBlock();
-		if (block == null) {
-			return;
-		}
-		if (block.getType() != Material.GOLD_BLOCK) {
-			return;
-		}
-		if (event.getPlayer().isSneaking()) {
+	public void goldBlockJump(PlayerJumpEvent event) {
+		Block below = event.getFrom().getBlock().getRelative(BlockFace.DOWN);
+		if (below.getType() != Material.GOLD_BLOCK) {
 			return;
 		}
-		//require player to stand on block
-		if (!event.getPlayer().getLocation().getBlock().getLocation().equals(block.getRelative(BlockFace.UP).getLocation())) {
-			return;
-		}
-		for(int y = block.getY() + 1; y <= 255; y++) {
-			if (doTeleport(block, event.getPlayer(), y)) {
+		for(int y = below.getY() + 1; y <= 255; y++) {
+			if (doTeleport(below, event.getPlayer(), y)) {
 				return;
 			}
 		}
-		for(int y = 0; y < block.getY(); y++) {
-			if (doTeleport(block, event.getPlayer(), y)) {
+		event.getPlayer().sendMessage(ChatColor.RED + "No suitable destination was found");
+	}
+
+	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+	public void goldBlockSneak(PlayerToggleSneakEvent event) {
```

Comment:
You should check if the player is enabling or disabling their sneak. If they're disabling it, you should return. If you don't, one press of shift will teleport the player twice.